scripts/build/termux_step_install_license.sh

```diff
@@ -60,29 +60,29 @@ termux_step_install_license() {
 						else
 							cp -f "${TERMUX_PKG_SRCDIR}/$FILE" "${TERMUX_PREFIX}/share/doc/${TERMUX_PKG_NAME}/copyright"
 						fi
-						# since this is a post-increment, (( 0 )) would be falsey
-						# thus `set -e` would kill the script on the first iteration
-						# using `true` prevents this
-						: $(( COUNTER++ ))
 					}
 				done
 				;;
 				# For the rest we can use a link to the generic license file
 				*) [[ -f "$TERMUX_SCRIPTDIR/packages/termux-licenses/LICENSES/${LICENSE}.txt" ]] && {
 						# the link target depends on the libc being used
-						if [[ "$TERMUX_PACKAGE_LIBRARY" == 'bionic' ]]; then
-							TO_LICENSE="../../LICENSES/${LICENSE}.txt"
-						elif [[ "$TERMUX_PACKAGE_LIBRARY" == 'glibc' ]]; then
-							TO_LICENSE="../../../../share/LICENSES/${LICENSE}.txt"
-						fi
+						case "$TERMUX_PACKAGE_LIBRARY" in
+							'bionic') TO_LICENSE="../../LICENSES/${LICENSE}.txt";;
+							'glibc')  TO_LICENSE="../../../../share/LICENSES/${LICENSE}.txt";;
+							*)        termux_error_exit "'$TERMUX_PACKAGE_LIBRARY' is not a supported libc";;
+						esac
 						if (( COUNTER )); then
-							ln -sf "$TO_LICENSE" "$TERMUX_PREFIX/share/doc/$TERMUX_PKG_NAME/copyright.$((COUNTER++))"
+							ln -sf "$TO_LICENSE" "$TERMUX_PREFIX/share/doc/$TERMUX_PKG_NAME/copyright.${COUNTER}"
 						else
 							ln -sf "$TO_LICENSE" "$TERMUX_PREFIX/share/doc/$TERMUX_PKG_NAME/copyright"
 						fi
+						continue
 					}
+					# If we get here no license file could be found
+					termux_error_exit "${TERMUX_PKG_NAME}: Could not find a license file for $LICENSE in packages/termux-licenses"
```

Comment:
@agnostic-apollo try this on for size?

Reply:
Could be done. @Grimler91 should know more about package specific license issues.

Reply:
Please take note that I've changed the way this check is handled a bit.

For the "specific" licenses there is now a `FOUND` value that gets incremented when a license file is found in source files.
If it is still `0` after all `COMMON_LICENSE_FILES` have been checked we exit with an error message about not being able to find a license for `$LICENSE`.

In the "generic" license case If there's no license file for this value of `$LICENSE` in `termux-licenses`, we return a missing license error.

Reply:
See the thread below for continued discussion.
https://github.com/termux/termux-packages/pull/21005#discussion_r1701120492